closed/src/java.base/share/classes/openj9/internal/security/RestrictedSecurityConfigurator.java

```diff
@@ -0,0 +1,325 @@
+
+                // Add restricted security Properties.
+                setProperties(props, restricts);
+
+                // Print out the Trace info.
+                if (userSecurityTrace) {
+                    restricts.listTrace();
+                }
+
+                if (debug != null) {
+                    debug.println("Restricted security mode loaded.");
+                    debug.println("Restricted security mode properties: " + props.toString());
+                }
+
+                securityEnabled = true;
+            }
+
+        } catch (Exception e) {
+            if (debug != null) {
+                debug.println("Unable to load restricted security mode configurations.");
+            }
+            e.printStackTrace();
+        }
+        return securityEnabled;
+    }
+
+    /**
+     * Load user restricted security settings from system property.
+     */
+    private static void initUserSetting() {
+
+        if (debug != null) {
+            debug.println("Loading user restricted security settings.");
+        }
+
+        String[] inputs = userSecuritySetting.split(",");
+
+        // For input ",,"
+        if (inputs.length == 0) {
+            new RuntimeException("user restricted security setting " + userSecuritySetting + " incorrect.")
+                    .printStackTrace();
+            System.exit(1);
+        }
+
+        for (String input : inputs) {
+            String in = input.trim();
+            if (in.equalsIgnoreCase("audit")) {
+                userSecurityAudit = true;
+            } else if (in.equalsIgnoreCase("help")) {
+                userSecurityHelp = true;
+            } else if (in.equalsIgnoreCase("trace")) {
+                userSecurityTrace = true;
+            } else {
+                try {
+                    userSecurityNum = Integer.parseInt(in);
+                } catch (NumberFormatException e) {
+                    new RuntimeException("user restricted security setting " + userSecuritySetting + " incorrect.")
+                            .printStackTrace();
+                    System.exit(1);
+                }
+            }
+        }
+
+        if (debug != null) {
+            debug.println("Loaded user restricted security settings, with userSecurityNum: "
+                    + String.valueOf(userSecurityNum) + " userSecurityTrace: " + userSecurityTrace
+                    + " userSecurityAudit: " + userSecurityAudit + " userSecurityHelp: " + userSecurityHelp);
+        }
+    }
+
+    /**
+     * Add restricted security providers.
+     *
+     * @param providers the provider name array
+     */
+    private static void setProviders(Properties props, ArrayList<String> providers) {
+
+        if (debug != null) {
+            debug.println("Adding restricted security provider.");
+        }
+
+        int pNum = 1;
+        for (String provider : providers) {
+            props.setProperty("security.provider." + pNum, provider);
+            pNum ++;
```

Comment:
Rather than start at 1, just increment before the use:
```java
        int pNum = 0;
        for (String provider : providers) {
            pNum += 1;
            props.setProperty("security.provider." + pNum, provider);
```